doc/_scripts/gen_boards_catalog.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2024 The Linux Foundation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from collections import namedtuple
+from pathlib import Path
+
+import list_boards
+import pykwalify
+import yaml
+import zephyr_module
+from gen_devicetree_rest import VndLookup
+
+ZEPHYR_BASE = Path(__file__).parents[2]
+
+logger = logging.getLogger(__name__)
+
+def guess_file_from_patterns(directory, patterns, name, extensions):
+    for pattern in patterns:
+        for ext in extensions:
+            matching_file = next(directory.glob(pattern.format(name=name, ext=ext)), None)
+            if matching_file:
+                return matching_file
+    return None
+
+
+def guess_image(board_or_shield):
+    img_exts = ["jpg", "jpeg", "webp", "png"]
+    patterns = [
+        "**/{name}.{ext}",
+        "**/*{name}*.{ext}",
+        "**/*.{ext}",
+    ]
+    img_file = guess_file_from_patterns(
+        board_or_shield.dir, patterns, board_or_shield.name, img_exts
+    )
+    return (Path("../_images") / img_file.name).as_posix() if img_file else ""
+
+
+def guess_doc_page(board_or_shield):
+    patterns = [
+        "doc/index.{ext}",
+        "**/{name}.{ext}",
+        "**/*{name}*.{ext}",
+        "**/*.{ext}",
+    ]
+    doc_file = guess_file_from_patterns(
+        board_or_shield.dir, patterns, board_or_shield.name, ["rst"]
+    )
+    return doc_file
+
+
+def get_catalog():
```

Comment:
note: why is this a standalone script if it does not have any entry point? It looks like a library, could well live within extensions?

Reply:
I will probably re-introduce an entry point as JSON/YAML export of the "catalog" might be useful for folks? In the meantime, if this is the only outstanding comment though would you consider the PR ready to go? :) Cheers!